pkg/sql/sem/tree/type_name.go

```diff
@@ -131,6 +131,17 @@ func ResolveType(
 		}
 		return types.MakeArray(typ), nil
 	case *UnresolvedObjectName:
+		// Account for geometry and geography being a PostgreSQL extension
+		// whose types live in the public schema.
+		if t.Schema() == string(PublicSchemaName) {
```

Comment:
I wouldn't put this here -- I think this should go in the `LogicalSchemaAccessor`